include/seqan3/range/view/translation.hpp

```diff
@@ -0,0 +1,159 @@
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \ingroup view
+ * \author Sara Hetzel <sara.hetzel AT fu-berlin.de>
+ * \brief Provides views for nucleotide to aminoacid translation.
+ */
+
+#pragma once
+
+#include <vector>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/alphabet/aminoacid/aa27.hpp>
+#include <seqan3/alphabet/translation/translation.hpp>
+#include <seqan3/alphabet/translation/translation_details.hpp>
+#include <seqan3/range/concept.hpp>
+#include <range/v3/all.hpp>
+
+namespace seqan3::view
+{
+
+// Complement view
+auto complement = [] (auto c)
+{
+    return complement_table<decltype(c)>::VALUE[c.to_rank()];
+};
+
+// Translate triplet view
+auto translate = [] (auto triplet)
+{
+    return translate_triplet(triplet[0], triplet[1], triplet[2]);
+};
+
+// Translate single frame
+auto translate_single = ranges::view::chunk(3)
+                        | ranges::view::take_while([] (auto chnk) { return ranges::size(chnk) == 3; })
+                        | ranges::view::transform(translate);
+
+// Translate multiple frames
+struct translate_fn
+{
+public:
+    template <typename range_type, typename size_type>
+    auto operator()(range_type && s, size_type const & num) const
+    {
+        std::vector<ranges::any_random_access_view<aa27> > frames;
```

Comment:
I am still not too happy with type-erasing to `ranges::any_random_access_view`, because it means we loose sized'ness, but I think there is nothing we can to about it now.